iocage_lib/ioc_destroy.py

```diff
@@ -79,147 +79,138 @@ def __stop_jails__(datasets, path=None, root=False, clean=False):
                         except OSError:
                             pass
 
-        for dataset in datasets.dependents:
-            if "jails" not in dataset.name:
+        for dataset in datasets:
+            if 'jails' not in dataset:
                 continue
 
-            if dataset.type != libzfs.DatasetType.FILESYSTEM:
+            if self.zfs_get_property(dataset, 'type') != 'filesystem':
                 continue
 
-            if dataset.properties["mountpoint"].value == 'legacy':
+            mountpoint = self.zfs_get_property(dataset, 'mountpoint')
+            if mountpoint == 'legacy':
                 continue
 
             # This is just to setup a replacement.
-            path = path.replace("templates", "jails")
+            path = path.replace('templates', 'jails')
 
             try:
-                uuid = dataset.name.partition(
-                    f"{path}")[1].rsplit("/", 1)[1]
+                uuid = dataset.partition(
+                    f'{path}')[1].rsplit('/', 1)[1]
             except IndexError:
                 # A RELEASE dataset
                 return
 
             # We want the real path now.
-            _path = dataset.properties["mountpoint"].value.replace("/root", "")
-            # It gives us a string that says "none", not terribly
-            # useful, fixing that.
-            _path = _path if _path != "none" else None
+            _path = mountpoint.replace('/root', '')
 
-            if (dataset.name.endswith(uuid) or root) and _path is not None:
-                iocage_lib.ioc_stop.IOCStop(uuid, _path, silent=True)
+            if (dataset.endswith(uuid) or root) and _path is not None:
+                with suppress(BaseException):
+                    # Can be missing/corrupt/whatever configuration
+                    # Since it's being nuked anyways, we don't care.
+                    iocage_lib.ioc_stop.IOCStop(uuid, _path, silent=True,
+                                                suppress_exception=True)
 
     def __destroy_leftovers__(self, dataset, clean=False):
         """Removes parent datasets and logs."""
-        uuid = dataset.name.rsplit("/root")[0].split("/")[-1]
-        snapshot = False
-
-        try:
-            path = dataset.properties["mountpoint"].value
+        uuid = dataset.rsplit('/root')[0].split('/')[-1]
```

Comment:
This would be related to the above one as well
`'vol1/iocage/jails/root/root'.rsplit('/root')[0].split('/')[-1]`